src/util/aggregator_types.F90

```diff
@@ -0,0 +1,418 @@
+    real(kind=real64), dimension(:,:), pointer :: source_data => null()
+  end type aggregator_real64_2d_t
+
+  type, extends(aggregator_t) :: aggregator_real64_3d_t
+    real(kind=real64), dimension(:,:,:), allocatable :: storage
+    real(kind=real64), dimension(:,:,:), pointer :: source_data => null()
+  end type aggregator_real64_3d_t
+
+contains
+
+  subroutine aggregator_handle_init(this)
+    class(aggregator_handle_t), intent(inout) :: this
+
+    call this%aggregator%init()
+
+  end subroutine aggregator_handle_init
+
+  subroutine aggregator_handle_accumulate(this)
+    class(aggregator_handle_t), intent(inout) :: this
+
+    call this%aggregator%accumulate()
+
+  end subroutine aggregator_handle_accumulate
+
+  subroutine aggregator_handle_normalise(this)
+    class(aggregator_handle_t), intent(inout) :: this
+
+    call this%aggregator%normalise()
+
+  end subroutine aggregator_handle_normalise
+
+  subroutine aggregator_handle_reset(this)
+    class(aggregator_handle_t), intent(inout) :: this
+
+    call this%aggregator%reset()
+
+  end subroutine aggregator_handle_reset
+
+  subroutine aggregator_init(this)
+    class(aggregator_t), intent(inout) :: this
+
+    select type (this)
+    type is (aggregator_int32_1d_t)
+      if (.not. allocated(this%storage)) allocate(this%storage, mold=this%source_data)
+    type is (aggregator_int32_2d_t)
+      if (.not. allocated(this%storage)) allocate(this%storage, mold=this%source_data)
+    type is (aggregator_int32_3d_t)
+      if (.not. allocated(this%storage)) allocate(this%storage, mold=this%source_data)
+    type is (aggregator_real32_1d_t)
+      if (.not. allocated(this%storage)) allocate(this%storage, mold=this%source_data)
+    type is (aggregator_real32_2d_t)
+      if (.not. allocated(this%storage)) allocate(this%storage, mold=this%source_data)
+    type is (aggregator_real32_3d_t)
+      if (.not. allocated(this%storage)) allocate(this%storage, mold=this%source_data)
+    type is (aggregator_real64_1d_t)
+      if (.not. allocated(this%storage)) allocate(this%storage, mold=this%source_data)
+    type is (aggregator_real64_2d_t)
+      if (.not. allocated(this%storage)) allocate(this%storage, mold=this%source_data)
+    type is (aggregator_real64_3d_t)
+      if (.not. allocated(this%storage)) allocate(this%storage, mold=this%source_data)
+    end select
+
+    call this%reset()
+
+  end subroutine aggregator_init
+
+  subroutine aggregator_set_method(this, method)
+    class(aggregator_t), intent(inout) :: this
+    character(len=*), intent(in) :: method
+
+    if (method == "mean") then
+      this%accumulate => sum_accumulate
+      this%normalise => mean_normalise
+      this%reset => other_reset
+    elseif (method == "sum") then
+      this%accumulate => sum_accumulate
+      this%normalise => other_normalise
+      this%reset => other_reset
+    elseif (method == "point") then
+      this%accumulate => point_accumulate
+      this%normalise => other_normalise
```

Comment:
This does not correspond to the subroutines it seems. In the subroutines, you have `point_accumulate` that doesn't do anything and `point_normalise` that attributes the value at the timestep.

I would rather do it as you have it here: `point_accumulate` attributes the current values, then`other_normalise` does nothing.

If we go with an incremental average, then we don't need any normalise step (all the aggregations use the `other_normalise` subroutine that doesn't do anything.